Week6/solution6.txt

```diff
@@ -0,0 +1,113 @@
+// CSV FILE PROGRAM
+1. Create a result table which contain student id, student name and 5 subject marks. 
+
+2. Enter 10 studnet details with its marks.
+
+3. Dump table into csv file "result.csv".
+
+4. Read result.csv file and Print Total Marks and Grade of each student. Also Append Total Marks and Grade column into result.csv file.
+
+5. List out Top 3 Student id and name with its percentage.
+
+SOLUTION:
+
+import pandas as pd
+
+data={'ID':[1,2,3,4,5,6,7,8,9,10],'NAME':['Heer','Hanee','Riya','Kishu','Hiya','Om','Sai','Ram','Shiv','Shankar'],'Sub1':[90,90,90,90,90,80,70,60,34,56],'Sub2':[90,79,90,60,30,80,87,45,67,10],'Sub3':[90,40,30,50,90,56,76,86,78,56],'Sub4':[90,50,90,90,90,67,56,76,66,78],'Sub5':[90,90,90,90,90,56,56,78,45,34]}
+
+df=pd.DataFrame(data)
+print(df)
+   ID     NAME  Sub1  Sub2  Sub3  Sub4  Sub5
+0   1     Heer    90    90    90    90    90
+1   2    Hanee    90    79    40    50    90
+2   3     Riya    90    90    30    90    90
+3   4    Kishu    90    60    50    90    90
+4   5     Hiya    90    30    90    90    90
+5   6       Om    80    80    56    67    56
+6   7      Sai    70    87    76    56    56
+7   8      Ram    60    45    86    76    78
+8   9     Shiv    34    67    78    66    45
+9  10  Shankar    56    10    56    78    34
+
+df.to_csv('c:\\sqlite3\\csv\\result.csv',index=False)
+
+df['TOTAL'],df['GRADE']=[[450,349,390,380,390,339,345,345,290,234],['A','B','B','B','B','C','C','C','D','D']]
+print(df)
+   ID     NAME  Sub1  Sub2  Sub3  Sub4  Sub5  TOTAL GRADE
+0   1     Heer    90    90    90    90    90    450     A
+1   2    Hanee    90    79    40    50    90    349     B
+2   3     Riya    90    90    30    90    90    390     B
+3   4    Kishu    90    60    50    90    90    380     B
+4   5     Hiya    90    30    90    90    90    390     B
+5   6       Om    80    80    56    67    56    339     C
+6   7      Sai    70    87    76    56    56    345     C
+7   8      Ram    60    45    86    76    78    345     C
+8   9     Shiv    34    67    78    66    45    290     D
+9  10  Shankar    56    10    56    78    34    234     D
+
+df.to_csv('c:\\sqlite3\\csv\\result.csv',index=False)
+
+num=df.nlargest(3,'TOTAL')
+print(num)
+   ID  NAME  Sub1  Sub2  Sub3  Sub4  Sub5  TOTAL GRADE
+0   1  Heer    90    90    90    90    90    450     A
+2   3  Riya    90    90    30    90    90    390     B
+4   5  Hiya    90    30    90    90    90    390     B
+
+
+
+
+//TEXT FILE PROGRAM 
+
+Program 1: Read Python.txt file and Print it in Reverse.
+import csv
+with open('c:\\sqlite3\\CSV\\python.txt','r',errors='ignore') as r:

```

Comment:
What is the purpose of errors=ignore

Reply:
when we want to skip the errors and countinue to run program then we use erros=ignore

Reply:
Ok..